ios/PurchasesHybridCommon/PurchasesHybridCommon/PaywallProxy.swift

```diff
@@ -32,11 +32,19 @@ import UIKit
             return
         }
 
-        let controller = PaywallViewController(displayCloseButton: true)
-        controller.delegate = self
-        controller.modalPresentationStyle = .pageSheet
+        let controller = PaywallViewController()
+        configureAndPresentPaywall(controller, rootController: rootController)
+    }
 
-        rootController.present(controller, animated: true)
+    @objc
+    public func presentPaywall(displayCloseButton: Bool) {
+        guard let rootController = UIApplication.shared.keyWindow?.rootViewController else {
+            NSLog("Unable to find root UIViewController")
+            return
+        }
+
+        let controller = PaywallViewController(displayCloseButton: displayCloseButton)
+        configureAndPresentPaywall(controller, rootController: rootController)
```

Comment:
Hmm considering this is very close to the other `presentPaywall`, could we create a private function with the optional `displayCloseButton` parameter instead?